src/library-authoring/components/ContainerCard.tsx

```diff
@@ -16,14 +17,73 @@ import { useSidebarContext } from '../common/context/SidebarContext';
 import { useLibraryRoutes } from '../routes';
 import BaseCard from './BaseCard';
 import messages from './messages';
+import DeleteModal from '../../generic/delete-modal/DeleteModal';
+import { ToastContext } from '../../generic/toast-context';
+import { useDeleteContainer, useRestoreContainer } from '../data/apiHooks';
 
 type ContainerMenuProps = {
   hit: ContainerHit,
 };
 
 const ContainerMenu = ({ hit } : ContainerMenuProps) => {
   const intl = useIntl();
-  const { contextKey, blockId } = hit;
+  const {
+    contextKey,
+    blockId,
+    usageKey: containerKey,
+    blockType: componentType,
+    displayName,
+  } = hit;
+  const {
+    sidebarComponentInfo,
+    closeLibrarySidebar,
+  } = useSidebarContext();
+
+  const [isConfirmingDelete, confirmDelete, cancelDelete] = useToggle(false);
+  const { showToast } = useContext(ToastContext);
+  const deleteContainerMutation = useDeleteContainer(containerKey);
+
+  let deleteWarningTitle;
+  let deleteText;
+  let deleteSuccess;
+  let deleteError;
+  if (componentType === 'unit') {
+    deleteWarningTitle = intl.formatMessage(messages.deleteUnitWarningTitle);
+    deleteText = intl.formatMessage(messages.deleteUnitConfirm, {
+      componentName: displayName,
+    });
+    deleteSuccess = intl.formatMessage(messages.deleteUnitSuccess);
+    deleteError = intl.formatMessage(messages.deleteUnitFailed);
+  }
+
+  const restoreContainerMutation = useRestoreContainer(containerKey);
+  const restoreComponent = useCallback(async () => {
+    try {
+      await restoreContainerMutation.mutateAsync();
+      showToast(intl.formatMessage(messages.undoDeleteContainerToastMessage));
+    } catch (e) {
+      showToast(intl.formatMessage(messages.undoDeleteUnitToastFailed));
```

Comment:
nit: this error message is unit-specific, while the others are Container messages -- should this message be set with the container-type-specific `const`s above too? Or can it be made more general for all container types?

Reply:
Updated: https://github.com/openedx/frontend-app-authoring/pull/1773/commits/a17967fc25c7a1a8165188b29b31666cb4c76391